pkg/domain/domain.go

```diff
@@ -2475,26 +2475,60 @@ func (do *Domain) autoAnalyzeWorker(owner owner.Manager) {
 	}
 }
 
-func (do *Domain) gcAnalyzeHistory(owner owner.Manager) {
-	defer util.Recover(metrics.LabelDomain, "gcAnalyzeHistory", nil, false)
+// analyzeJobsCleanupWorker is a background worker that periodically performs two main tasks:
+//  1. Garbage Collection: It removes outdated analyze jobs from the statistics handle.
+//     This operation is performed every hour and only if the current instance is the owner.
+//     Analyze jobs older than 7 days are considered outdated and are removed.
+//  2. Cleanup: It cleans up corrupted analyze jobs. This operation is performed every three stats leases.
+//     It first retrieves the list of current analyze processes, then removes any analyze job
+//     that is not associated with a current process.
+//     This operation is also performed only if the current instance is the owner.
+func (do *Domain) analyzeJobsCleanupWorker(owner owner.Manager) {
+	defer util.Recover(metrics.LabelDomain, "analyzeJobsCleanupWorker", nil, false)
+	// For GC.
 	const gcInterval = time.Hour
-	statsHandle := do.StatsHandle()
+	const DaysToKeep = 7
 	gcTicker := time.NewTicker(gcInterval)
+	// For clean up.
+	// Default stats lease is 3 * time.Second.
+	// So cleanupInterval is 100 * 3 * time.Second = 5 * time.Minute.
+	var cleanupInterval = do.statsLease * 100
+	cleanupTicker := time.NewTicker(cleanupInterval)
 	defer func() {
 		gcTicker.Stop()
-		logutil.BgLogger().Info("gcAnalyzeHistory exited.")
+		cleanupTicker.Stop()
+		logutil.BgLogger().Info("analyzeJobsCleanupWorker exited.")
 	}()
+	statsHandle := do.StatsHandle()
 	for {
 		select {
 		case <-gcTicker.C:
+			// Only the owner should perform this operation.
 			if owner.IsOwner() {
-				const DaysToKeep = 7
 				updateTime := time.Now().AddDate(0, 0, -DaysToKeep)
 				err := statsHandle.DeleteAnalyzeJobs(updateTime)
 				if err != nil {
 					logutil.BgLogger().Warn("gc analyze history failed", zap.Error(err))
 				}
 			}
+		case <-cleanupTicker.C:
+			sm := do.InfoSyncer().GetSessionManager()
+			if sm == nil {
+				continue
+			}
+			analyzeProcessIDs := make(map[uint64]struct{}, 8)
+			for _, process := range sm.ShowProcessList() {
+				if strings.HasPrefix(strings.ToLower(process.Info), "analyze table") {
+					analyzeProcessIDs[process.ID] = struct{}{}
+				}
+			}
+			// Only the owner should perform this operation.
+			if owner.IsOwner() {
```

Comment:
it can be moved into ```owner.IsOwner()```
```suggestion
			analyzeProcessIDs := make(map[uint64]struct{}, 8)
			
			// Only the owner should perform this operation.
			if owner.IsOwner() {
			    for _, process := range sm.ShowProcessList() {
				    if strings.HasPrefix(strings.ToLower(process.Info), "analyze table") {
					    analyzeProcessIDs[process.ID] = struct{}{}
				    }
			    }
```

Reply:
Good catch. updated.